src/modules/CaaSMapper.ts

```diff
@@ -737,26 +827,50 @@ export class CaaSMapper {
               remoteProject: remoteProjectId,
               filterContext,
             },
-            this.referenceDepth
-          )
-        )
-      )
-      const fetchedItems = response.map(({ items }) => items).flat()
-      ids.forEach((id) =>
-        referencedItems[id].forEach((path) =>
-          set(
-            data,
-            path,
-            fetchedItems.find((data) => {
-              const hasId = typeof data === 'object' && 'id' in (data as object)
-              if (hasId) {
-                return (data as { id: string }).id === id
-              }
-            })
+            this
           )
         )
       )
     }
-    return data
+  }
+
+  // we use a query function, because ids get mixed up:
+  // referencedItems store identifier, resolvedReferences store _id or previewId
+  static findResolvedReferencesByIds(ids: string[], resolvedReferences: ResolvedReferencesInfo) {
+    const items = []
+    for (const [resolvedId, resolvedItem] of Object.entries(resolvedReferences)) {
+      for (const id of ids) {
+        if (id.includes(resolvedId) || resolvedId.includes(id)) {
```

Comment:
This would be also be simpler if we would assure usage of _id and previewId consistently